org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/TOSCAFileValidator.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+package org.eclipse.winery.lsp.Server.ServerCore;
+
+import org.yaml.snakeyaml.error.Mark;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+
+public class TOSCAFileValidator {
+    public ArrayList<TOSCAFileDiagnostics> diagnostics = new ArrayList<>();
+
+    public void validateRequiredKeys(Map<String, Object> yamlMap, Path path) {
+        if (!yamlMap.containsKey("tosca_definitions_version")) {
```

Comment:
i think, this is too much effort.

You need to convert the YAML to Java Objects in all cases.

If you use `https://github.com/Randgalt/record-builder`, it should throw an exception of you use `null` for a field.

Meaning: If you access the yamlMap for a key you need - this also throws an NPE.

Thus, the conversion is "just" load the YAML in YAML structure and then go through the YAML structure with java code. thereby, you generate the TOSCA objects.

Reply:
I have tried to do so the problem is that if there are two null fields it will only throw NPE for the first field, which means that I would not be able to catch all diagnoses.